pageserver/src/tenant.rs

```diff
@@ -3098,6 +3091,12 @@ impl Tenant {
             }
         }
 
+        // Don't do compactions other than L0 compaction if the circuit breaker is tripped.
+        if self.compaction_circuit_breaker.lock().unwrap().is_broken() {
+            info!("skipping compaction due to previous failures");
+            return Ok(CompactionOutcome::Skipped);
+        }
```

Comment:
Shouldn't L0 compaction be subject to the circuit breaker too, to avoid doing a bunch of wasted work?

The compaction loop does have a 300s cooldown on errors though, which is probably sufficient considering there's a bound on how much wasted work we'd do during L0 compaction anyway. It's just not clear to me why we would special-case image compaction here.

We could even just remove the circuit breaker entirely and use a larger backoff in the compaction loop instead.

Reply:
I removed changes to the circuit breaker in this patch